include/AST/Types/ArrayTy.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef GLU_AST_TYPES_ARRAYTY_HPP_
+#define GLU_AST_TYPES_ARRAYTY_HPP_
+
+#include "Types/TypeBase.hpp"
+#include <optional>
+
+namespace glu::types {
+
+/// @brief Represents an array type.
+class ArrayTy : public TypeBase {
+    TypeKind const _dataKind;
```

Comment:
this should be a `TypeBase *`
we don't want to store whether an array contains another array, we want to store that an array contains an array of ints or something